src/common/dom/apply_themes_on_element.ts

```diff
@@ -37,6 +38,11 @@ export const applyThemesOnElement = (
     if (themeOptions.dark) {
       cacheKey = `${cacheKey}__dark`;
       themeRules = darkStyles;
+      themeRules["dark-app-header-background-color"] = blend(
+        themeOptions.primaryColor || "#03a9f4",
+        "#111",
+        15
+      );
```

Comment:
Move this in the `if (themeOptions.primaryColor)` block
```suggestion
      themeRules["app-header-background-color"] = blend(
        themeOptions.primaryColor || "#03a9f4",
        "#111",
        15
      );
```

Reply:
Added both commits, but moving this outside of `if (themeOptions.dark) {` causes some strange behaviors. When selecting a new color while in the dark mode profile settings it will change to that new color (say orange), but then pressing reset does nothing (stays orange), change the color again to something new and then hit reset and it will reset to the first selected color (orange) rather than the default blue color.

Reply:
Keeping it in `if (themeOptions.dark)` is the only solution I've found so far, but I'll keep testing.

Reply:
It should only be applied if both dark mode and primary color is set

Reply:
Ah, okay. So moving it to the suggested "if" but adding a if (themeOptions.dark) might do the trick.